autotest/gdrivers/gdalhttp.py

```diff
@@ -46,127 +46,87 @@
     sys.path.append('../pymod')
     import gdaltest
 
-###############################################################################
-# Verify we have the driver.
 
+pytestmark = pytest.mark.require_driver('HTTP')
 
-def test_http_1():
 
-    gdaltest.dods_drv = None
+@pytest.fixture(autouse=True, scope="module")
+def disable_dods_driver():
+    dods_drv = gdal.GetDriverByName('DODS')
+    if dods_drv is not None:
+        dods_drv.Deregister()
+    yield
+    if dods_drv is not None:
+        dods_drv.Register()
+    dods_drv = None
 
-    drv = gdal.GetDriverByName('HTTP')
-    if drv is None:
-        pytest.skip()
 
-    gdaltest.dods_drv = gdal.GetDriverByName('DODS')
-    if gdaltest.dods_drv is not None:
-        gdaltest.dods_drv.Deregister()
+@pytest.fixture(autouse=True)
+def set_http_timeout():
+    with gdaltest.config_option('GDAL_HTTP_TIMEOUT', '5'):
+        yield
 
-    tst = gdaltest.GDALTest('PNG', 'http://gdal.org/gdalicon.png',
-                            1, 7617, filename_absolute=1)
-    ret = tst.testOpen()
-    if ret == 'fail':
-        conn = gdaltest.gdalurlopen('http://gdal.org/gdalicon.png')
-        if conn is None:
-            pytest.skip('cannot open URL')
-        conn.close()
-
-    return ret
 
-###############################################################################
-# Verify /vsicurl (subversion file listing)
+def skip_if_unreachable(url, try_read=False):
+    conn = gdaltest.gdalurlopen(url, timeout=4)
+    if conn is None:
+        pytest.skip('cannot open URL')
+    if try_read:
+        try:
+            conn.read()
+        except Exception:
+            pytest.skip('cannot read')
+    conn.close()
 
 
-def test_http_2():
+###############################################################################
+# Verify we have the driver.
 
-    drv = gdal.GetDriverByName('HTTP')
-    if drv is None:
-        pytest.skip()
 
-    tst = gdaltest.GDALTest('GTiff', '/vsicurl/https://raw.githubusercontent.com/OSGeo/gdal/release/3.1/autotest/gcore/data/byte.tif',
-                            1, 4672, filename_absolute=1)
+def test_http_1():
+    url = 'http://gdal.org/gdalicon.png'
+    tst = gdaltest.GDALTest('PNG', url, 1, 7617, filename_absolute=1)
     ret = tst.testOpen()
     if ret == 'fail':
-        conn = gdaltest.gdalurlopen('https://raw.githubusercontent.com/OSGeo/gdal/release/3.1/autotest/gcore/data/byte.tif')
-        if conn is None:
-            pytest.skip('cannot open URL')
-        conn.close()
-
-    return ret
+        skip_if_unreachable(url)
+        pytest.fail()
```

Comment:
Tests 1 and 2 previously ended with `return ret`, but I've not been able to find any evidence that pytest would parse the result of `ret`. I've changed them to explicitly fail, in line with the other tests.